functions/Remove-DbaDbView.ps1

```diff
@@ -0,0 +1,150 @@
+
+    .PARAMETER View
+        The view(s) to process. If unspecified, all views will be processed.
+
+    .PARAMETER IncludeSystemDbs
+        If this switch is enabled, views can be removed from system databases.
+
+    .PARAMETER InputObject
+        Enables piped input from Get-DbaDbView or Get-DbaDatabase
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: View, Database
+        Author: Mikey Bronowski (@MikeyBronowski), https://bronowski.it
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/Remove-DbaDbView
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbView -SqlInstance localhost -Database dbname -View "customview1", "customview2"
+
+        Removes views customview1 and customview2 from the database dbname on the local default SQL Server instance.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbView -SqlInstance localhost, sql2016 -Database db1, db2 -View view1, view2, view3
+
+        Removes view1, view2, view3 from db1 and db2 on the local and sql2016 SQL Server instances.
+
+    .EXAMPLE
+        PS C:\> Remove-DbaDbView -SqlInstance localhost -Database msdb -IncludeSystemDbs -View view1, view2, view3
+
+        Removes view1, view2, view3 from db1 and db2 from system database (msdb) on the local SQL Server instances.
+
+    .EXAMPLE
+        PS C:\> $servers = Get-Content C:\servers.txt
+        PS C:\> $servers | Remove-DbaDbView -Database db1, db2 -View view1
+
+        Removes view1 from db1 and db2 on the servers in C:\servers.txt file.
+
+    .EXAMPLE
+        PS C:\> $views = Get-DbaDbView -SqlInstance localhost, sql2016 -Database db1, db2 -View view1, view2, view3
+        PS C:\> $views | Remove-DbaDbView
+
+        Removes view1, view2, view3 from db1 and db2 on the local and sql2016 SQL Server instances.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'High')]
+    param (
+        [parameter(ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [string[]]$View,
+        [switch]$IncludeSystemDbs,
+        [parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
```

Comment:
```suggestion
        [Microsoft.SqlServer.Management.Smo.View[]]$InputObject,
```

This resolves any need to having to check what type is the input object. The object we output from `Get-DbaDbView` includes SqlInstance name and PowerShell will bind that for us without any hassle or extra code.

https://github.com/sqlcollaborative/dbatools/blob/de024aaa616fe37de7c8eedb796a2352ab08529a/functions/Get-DbaDbView.ps1#L167-L175

Reply:
Would that work just remove- standalone, without piping from get-

I would probably use get | remove myself as "safer" option anyway.

Reply:
If you are calling remove function by itself you don't use the input object parameter.

Reply:
Do my changes make more sense now?